android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginScreen.kt

```diff
@@ -71,81 +132,140 @@ fun LoginScreenInner(
                     modifier = Modifier.size(280.dp),
                 )
             }
-            Row(modifier = Modifier.weight(1f).fillMaxSize()) {
-                LoginForm(email, password, onEmailChange, onPasswordChange, onLoginPress)
-            }
+            Row(modifier = Modifier.weight(1f).fillMaxSize()) { LoginForm(state, events, 280.dp) }
         }
     }
 }
 
 @Composable
-fun LoginForm(
-    email: String,
-    password: String,
-    onEmailChange: (String) -> Unit,
-    onPasswordChange: (String) -> Unit,
-    onLoginPress: () -> Unit,
+private fun LoginForm(
+    state: LoginState,
+    events: LoginEvents,
+    width: Dp,
     modifier: Modifier = Modifier,
 ) {
     Column(
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.spacedBy(4.dp),
         modifier = modifier.fillMaxSize(),
     ) {
-        EmailField(email, onEmailChange)
-        PasswordField(password, onPasswordChange)
-        Row(modifier = Modifier.width(280.dp), horizontalArrangement = Arrangement.spacedBy(4.dp)) {
-            RegisterButton(modifier = Modifier.weight(1f)) {}
-            LoginButton(modifier = Modifier.weight(1f), onLoginPress)
+        LoginFields(state, events, Modifier.width(width))
+        AnimatedVisibility(!state.registering) {
```

Comment:
Can we hide the password field when we use the forgot password option? I think this might involve adding an extra state to our page or making an entirely new UI page